src/components/dev-hub/champion-application-form.js

```diff
@@ -0,0 +1,293 @@
+const SUCCESS_STATE_HEADING_MOBILE_LINE_HEIGHT = '38px';
+const MODAL_DIALOG_CONTAINER_STYLE_WIDTH = '664px';
+
+const ModalContainer = styled('div')`
+    @media ${screenSize.mediumAndUp} {
+        padding: 0 ${size.mediumLarge};
+    }
+`;
+
+const contentStyle = css`
+    @media ${screenSize.upToMedium} {
+        padding: ${size.default} ${size.default}
+            ${CONTENT_STYLE_MOBILE_BOTTOM_PADDING};
+    }
+    @media ${screenSize.mediumAndUp} {
+        padding: ${size.default} ${size.default} ${CONTENT_STYLE_BOTTOM_PADDING};
+    }
+`;
+
+const ButtonContainer = styled('div')`
+    display: flex;
+    justify-content: center;
+`;
+
+const Title = styled(H5)`
+    margin-bottom: ${size.mediumLarge};
+    @media ${screenSize.upToMedium} {
+        font-size: ${fontSize.medium};
+        line-height: ${TITLE_MOBILE_LINE_HEIGHT};
+    }
+`;
+
+const Description = styled(P)`
+    margin-bottom: ${DESCRIPTION_BOTTOM_MARGIN};
+    @media ${screenSize.upToMedium} {
+        margin-bottom: ${size.large};
+    }
+`;
+
+const StyledInput = styled(Input)`
+    margin-bottom: ${size.mediumLarge};
+`;
+
+const ErrorMessage = styled(P)`
+    color: ${({ theme }) => theme.colorMap.salmon};
+`;
+
+const NameSection = styled('div')`
+    display: grid;
+    grid-template-columns: repeat(auto-fill, ${NAME_INPUT_FIELD_WIDTH});
+    justify-content: space-between;
+    @media ${screenSize.upToMedium} {
+        grid-template-columns: 100%;
+    }
+`;
+
+const StyledCheckbox = styled(Checkbox)`
+    /* Leafygreen's Checkbox input is given height/width of 0, hiding invalid popup */
+    > input {
+        height: ${size.medium};
+        width: ${size.medium};
+    }
+    margin-bottom: ${size.large};
+    @media ${screenSize.upToMedium} {
+        margin-bottom: ${CHECKBOX_MOBILE_BOTTOM_MARGIN};
+    }
+`;
+
+const RadioGroupTitle = styled(P3)`
+    margin-bottom: ${size.xsmall};
+    @media ${screenSize.upToMedium} {
+        font-size: ${fontSize.tiny};
+        line-height: ${lineHeight.tiny};
+    }
+`;
+
+const StyledRadioGroup = styled(RadioGroup)`
+    column-gap: ${size.mediumLarge};
+    display: grid;
+    grid-auto-flow: column;
+    grid-template-columns: repeat(3, max-content);
+    grid-template-rows: auto auto;
+    margin-bottom: ${size.mediumLarge};
+    max-width: 100%;
+    @media ${screenSize.upToMedium} {
+        grid-auto-flow: row;
+        grid-template-columns: auto;
+    }
+`;
+
+const StyledTextArea = styled(TextArea)`
+    margin-bottom: ${size.large};
+    @media ${screenSize.upToMedium} {
+        margin-bottom: ${size.default};
+    }
+`;
+
+const StyledSuccessState = styled(SuccessState)`
+    margin-bottom: ${SUCCESS_STATE_BOTTOM_MARGIN};
+    h3 {
+        @media ${screenSize.upToMedium} {
+            font-size: ${SUCCESS_STATE_HEADING_MOBILE_FONT_SIZE};
+            line-height: ${SUCCESS_STATE_HEADING_MOBILE_LINE_HEIGHT};
+        }
+    }
+`;
+
+const Form = React.memo(({ setSuccess, success }) => {
```

Comment:
Also I see we are going `React.memo` here. Any specific reason?

Reply:
> I also saw in my editor that `setSuccess` is not being used. This is to be used in the submit handler right?

thats right! will be used in my next PR!


> Also I see we are going `React.memo` here. Any specific reason?

I noticed the academia and project sign up forms using `React.memo`. From my understanding, it tells the `Form` to only re-render when the props change? I think it would help improve performance here too?